web/src/components/apps/AppLicense.tsx

```diff
@@ -35,51 +35,50 @@ type State = {
   messageType: string;
   showNextStepModal: boolean;
   entitlementsToShow: string[];
-  showLicenseChangeModal: boolean;
+  showLicenseChangeModalState: boolean;
   licenseChangeFile: LicenseFile | null;
   changingLicense: boolean;
   licenseChangeMessage: string;
   licenseChangeMessageType: string;
   isViewingLicenseEntitlements: boolean;
 };
 
-class AppLicense extends Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-
-    this.state = {
+const AppLicense = (props: Props) => {
+  const [state, setState] = useReducer(
+    (currentState: State, newState: Partial<State>) => ({
+      ...currentState,
+      ...newState,
+    }),
+    {
       appLicense: null,
       loading: false,
       message: "",
       messageType: "info",
       showNextStepModal: false,
       entitlementsToShow: [],
-      showLicenseChangeModal: false,
+      showLicenseChangeModalState: false,
```

Comment:
both state variable and function are named `showLicenseChangeModal` so I opted to update the state name to remove the conflict. TypeChecks made this super easy